### ISSUE TITLE
DISABLED test_backward_node_failure (__main__.TensorPipeAgentDistAutogradTestWithSpawn)

### ISSUE THREAD

**Issue by mrshenli** [pytorch/pytorch#39457]
https://app.circleci.com/pipelines/github/pytorch/pytorch/176463/workflows/013c36ff-c568-4726-a10f-fc6fc342ac0c/jobs/5670885/steps

```
Jun 03 17:38:08   test_backward_node_failure (__main__.TensorPipeAgentDistAutogradTestWithSpawn) ... [W tensorpipe_agent.cpp:312] RPC agent for worker3 encountered error when reading incoming request: pipe closed
Jun 03 17:38:08 [E container.cpp:248] Could not release Dist Autograd Context on node 0: pipe closed
Jun 03 17:38:08 [W tensorpipe_agent.cpp:312] RPC agent for worker0 encountered error when reading incoming request: EOF: end of file
Jun 03 17:38:08 [W tensorpipe_agent.cpp:280] RPC agent for worker0 encountered error when writing outgoing response: EOF: end of file
Jun 03 17:38:08 [W tensorpipe_agent.cpp:436] RPC agent for worker0 encountered error when writing outgoing request: ECONNREFUSED: connection refused
Jun 03 17:38:08 [E container.cpp:248] Could not release Dist Autograd Context on node 2: pipe closed
Jun 03 17:38:08 [W tensorpipe_agent.cpp:312] RPC agent for worker1 encountered error when reading incoming request: pipe closed
Jun 03 17:38:08 [W tensorpipe_agent.cpp:312] RPC agent for worker2 encountered error when reading incoming request: EOF: end of file
Jun 03 17:38:08 [W tensorpipe_agent.cpp:436] RPC agent for worker2 encountered error when writing outgoing request: ECONNREFUSED: connection refused
Jun 03 17:38:08 [W tensorpipe_agent.cpp:280] RPC agent for worker2 encountered error when writing outgoing response: EOF: end of file
Jun 03 17:38:08 [W tensorpipe_agent.cpp:436] RPC agent for worker2 encountered error when writing outgoing request: ECONNREFUSED: connection refused
Jun 03 17:38:08 [W tensorpipe_agent.cpp:453] RPC agent for worker0 encountered error when reading incoming request: EOF: end of file
Jun 03 17:38:08 [W tensorpipe_agent.cpp:436] RPC agent for worker2 encountered error when writing outgoing request: EPIPE: broken pipe
Jun 03 17:38:08 [W tensorpipe_agent.cpp:436] RPC agent for worker2 encountered error when writing outgoing request: EPIPE: broken pipe
Jun 03 17:38:08 [W tensorpipe_agent.cpp:436] RPC agent for worker0 encountered error when writing outgoing request: ECONNREFUSED: connection refused
Jun 03 17:38:08 [W tensorpipe_agent.cpp:436] RPC agent for worker2 encountered error when writing outgoing request: ECONNREFUSED: connection refused
Jun 03 17:38:08 [W tensorpipe_agent.cpp:436] RPC agent for worker2 encountered error when writing outgoing request: EPIPE: broken pipe
Jun 03 17:38:08 [W tensorpipe_agent.cpp:436] RPC agent for worker0 encountered error when writing outgoing request: EOF: end of file
Jun 03 17:38:08 [W tensorpipe_agent.cpp:436] RPC agent for worker0 encountered error when writing outgoing request: ECONNREFUSED: connection refused
Jun 03 17:38:08 [W tensorpipe_agent.cpp:436] RPC agent for worker0 encountered error when writing outgoing request: EOF: end of file
Jun 03 17:39:47 Timing out after 100 seconds and killing subprocesses.
Jun 03 17:39:47 ERROR (100.070s)
```

```
Jun 03 17:43:08 ======================================================================
Jun 03 17:43:08 ERROR [100.070s]: test_backward_node_failure (__main__.TensorPipeAgentDistAutogradTestWithSpawn)
Jun 03 17:43:08 ----------------------------------------------------------------------
Jun 03 17:43:08 Traceback (most recent call last):
Jun 03 17:43:08   File "/opt/conda/lib/python3.8/site-packages/torch/testing/_internal/common_distributed.py", line 204, in wrapper
Jun 03 17:43:08     self._join_processes(fn)
Jun 03 17:43:08   File "/opt/conda/lib/python3.8/site-packages/torch/testing/_internal/common_distributed.py", line 306, in _join_processes
Jun 03 17:43:08     self._check_return_codes(elapsed_time)
Jun 03 17:43:08   File "/opt/conda/lib/python3.8/site-packages/torch/testing/_internal/common_distributed.py", line 344, in _check_return_codes
Jun 03 17:43:08     raise RuntimeError('Process {} terminated or timed out after {} seconds'.format(i, elapsed_time))
Jun 03 17:43:08 RuntimeError: Process 2 terminated or timed out after 100.05238389968872 seconds
```

cc @ezyang @gchanan @zou3519 @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @rohan-varma @xush6528 @jjlilley @osalpekar @jiayisuse @lw @beauby

**Comment by mrshenli** [comment]
@lw I saw the following log appears in all tests. Is this expected?

```
Jun 03 17:38:07 [W tensorpipe_agent.cpp:312] RPC agent for worker0 encountered error when reading incoming request: pipe closed
```

**Comment by lw** [comment]
For now it is expected. There's a fixme in the code, but I haven't gotten to address it yet:
https://github.com/pytorch/pytorch/blob/ec5d579929b2c56418aacaec0874b92937d095a4/torch/csrc/distributed/rpc/tensorpipe_agent.cpp#L308-L316

It's harmless though.

**Comment by lw** [comment]
Might get fixed by #39535...

**Comment by mrshenli** [comment]
Closing this issue for now to enable this test and check if it still fails after #39535
